Sources/X509/Verifier/VerifierPolicy.swift

```diff
@@ -27,11 +64,22 @@ public enum PolicyEvaluationResult: Sendable {
 // Additionally, we should add conditional Sendable, Equatable, and Hashable conformances as needed.
 // This will also allow equivalent conditional conformances on `Verifier`.
 public struct PolicySet: VerifierPolicy {
+    public let processedExtensions: [ASN1ObjectIdentifier]
+
     @usableFromInline var policies: [any VerifierPolicy]
 
     @inlinable
     public init(policies: [any VerifierPolicy]) {
         self.policies = policies
+
+        var extensions: [ASN1ObjectIdentifier] = []
+        extensions.reserveCapacity(policies.reduce(into: 0, { $0 += $1.processedExtensions.count }))
+
+        for policy in policies {
+            extensions.append(contentsOf: policy.processedExtensions)
+        }
+
+        self.processedExtensions = extensions
```

Comment:
hm... we compute that upfront and not on demand. Therefore a `VerifierPolicy` can never check any policies dynamically e.g. it can't verify an extension only if another extensions is present as well. Is this on purpose? If it is on purpose, I think we should rename this property to reflect that this is not about the extensions it has processed but which it promises to process

Reply:
If we don't make the list a `Set` we could also just make it a function
```swift
protocol VerifierPolicy {
    func willVerifyCiritalExtension(_ extension: ASN1ObjectIdentifier) -> Bool
}
```
If we want to keep the property we can call it `var verifyingCriticalExtensions: [ASN1ObjectIdentiifer]` 

I kinda want to make it a static func/property but this will only work if `PolicySet` is a variadic generic type.